### DISCUSSION TITLE
METviewer container for HPC

### DISCUSSION THREAD

**Discussion by jwraby** [dtcenter/METplus#1679] [Plot Generation] [answered]
This discussion was suggested by John Halley Gotway during the METplus User's Workshop:
I would like to access the METviewer Docker container so I can pull and convert it into a Singularity container on the HPC. Past attempts to run a METviewer Singularity container on the HPC (SEP 2021) failed because it requires root access inside the container per the diagnosis from the HPC system administrator. Perhaps there have been new developments which will now enable the METviewer container to run. Thanks.

**Comment by CPKalb** [comment, marked as answer]
Hi:

There is an active development task on converting the METviewer container to singularity.  Information on the task can be found here: https://github.com/NCAR/container-dtc-nwp/issues/23.  It looks like these changes are close but not ready quite yet.

Christina

**Reply by jwraby** [reply]
Thanks for the update.

**Reply by JohnHalleyGotway** [reply]
@michelleharrold is the person who's been working with @TatianaBurek on this. @michelleharrold can you please provide @jwraby with some details about this? Perhaps you could point him to the DTC NWP containers tutorial once your updates are complete?

**Comment by michelleharrold** [comment]
Hi @jwraby,

As @CPKalb mentioned, we have a DTC project that is actively looking to run METviewer via Singularity. We have made substantial progress, but we haven't quite crossed the finish line. @TatianaBurek created a Dockerfile that builds an image that contains MariaDB and METviewer. We have successfully run and tested on AWS, but we haven't been able to get it working on NCAR's Cheyenne HPC. In addition, we are still in the process of updating our online tutorial and documentation.

If you are interested in taking a look, we have the necessary Dockerfile and additional scripts needed to get up and running on [GitHub](https://github.com/NCAR/container-dtc-nwp/tree/main/components/metviewer/METviewer).

In addition, we are populating our [project website with instructions](https://dtcenter.org/nwp-containers-online-tutorial/hurricane-sandy-case-27-oct-2012/visualize-verification-results) (but it is still under constructions with regard to having pushed our image to Docker Hub, e.g.). We will keep you posted as we make continual progress.

Thanks!
Michelle

**Reply by jwraby** [reply]
Thanks for the update, Michelle. Good progress!
R/
John
